### PR TITLE
updated filter model and method

### PR DESCRIPTION
**Pull request by GiftLanga**

### CODE REVIEW

File: WaterLog_Backend/Controllers/SegmentEventsController.cs [hunk 1]
```diff
@@ -68,16 +68,23 @@ public async Task AddSegmentEvent([FromBody] SegmentEventsEntry value)
             {
                 DateTime tempTime = DateTime.Now;
                 DateTime returnV = new DateTime(tempTime.Year, tempTime.Month, tempTime.Day, i, 0, 0);
```
Comment (ash-rabbit) [inline WaterLog_Backend/Controllers/SegmentEventsController.cs]: So everything looks good but your naming conventions are making it hard to read. Rename `returnV` and in your for loop you have `ret` maybe rename that too.

File: WaterLog_Backend/Controllers/SegmentEventsController.cs [hunk 2]
```diff
@@ -190,290 +197,55 @@ public async Task<List<GetAlerts>> GetAlertsByPage(int id = 1)
...
-                            );
-                        }
-                    }
-                }
-            } // filter faulty sensor end
-            return (ListOfAlerts.OrderByDescending(a => a.Date).OrderByDescending(a => a.Status).ToList());
+                return alerts
+                     .Where(alert => alert.Severity.ToLower() == filter.Severity.ToLower())
+                     .ToList();
+            } else if (filteredAlerts.Count != 0)
```
Comment (ash-rabbit) [inline WaterLog_Backend/Controllers/SegmentEventsController.cs]: put elseIf on the second line to bring consistency

File: WaterLog_Backend/Controllers/SegmentEventsController.cs [hunk 3]
```diff
@@ -190,290 +197,55 @@ public async Task<List<GetAlerts>> GetAlertsByPage(int id = 1)
...
-                                     litresUsed,
-                                     totalSystemLitres,
-                                     entry.ResolvedStatus
-                                  )
-                             );
-                        }
-                    }
+                    onlySeverity = false;
+                    sensorData = alerts
+                        .Where(alert => alert.EntityName.ToLower() == "tank" && alert.EntityId == filter.SensorId)
```
Comment (ash-rabbit) [inline WaterLog_Backend/Controllers/SegmentEventsController.cs]: add entity name to an enum instead of giving string value here to "tank" and "sensor"

**Review by ash-rabbit — CHANGES_REQUESTED** [review]

File: WaterLog_Backend/Controllers/SegmentEventsController.cs (hunk 1 above)
Comment (GiftLanga) [reply WaterLog_Backend/Controllers/SegmentEventsController.cs]: renamed variables

File: WaterLog_Backend/Controllers/SegmentEventsController.cs (hunk 2 above)
Comment (GiftLanga) [reply WaterLog_Backend/Controllers/SegmentEventsController.cs]: brought consistency

File: WaterLog_Backend/Controllers/SegmentEventsController.cs (hunk 3 above)
Comment (GiftLanga) [reply WaterLog_Backend/Controllers/SegmentEventsController.cs]: replaced string with entity name

**Review by ash-rabbit — APPROVED** [review]

**Pull request merged**